src/readii/io/loaders/general.py

```
@@ -0,0 +1,78 @@
+        else:
+            raise ValueError("Unsupported file format. Please provide a .csv or .xlsx file.")
+        
+        if df.empty:
+            raise ValueError("Loaded DataFrame is empty")
+        
+        return df
+    
+    except pd.errors.EmptyDataError:
+        raise ValueError("File is empty")
+    except (pd.errors.ParserError, ValueError) as e:
+        raise ValueError(f"Error parsing file: {e}")
```

Ruff (0.8.0)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)